coordinator/src/node.rs

```diff
@@ -140,6 +143,19 @@ impl Node {
         let (fee_payment_hash, invoice) = self.fee_invoice_taker(trade_params).await?;
         let mut connection = self.pool.get()?;
 
+        let order_id = trade_params.filled_with.order_id;
+        let order = orders::get_with_id(&mut connection, order_id)?
+            .with_context(|| "Could not find order with id {order_id}")?;
+
+        ensure!(
+            order.expiry > OffsetDateTime::now_utc(),
+            "Can't execute expired match"
+        );
+        ensure!(
+            order.order_state == OrderState::Match,
+            "Can't execute trade with in invalid state"
```

Comment:
Could be helpful to find future bugs if we print in what state the order was instead.